Tasks/Common/TlsHelper_/TlsHelper_.psm1

```diff
@@ -0,0 +1,55 @@
+Import-VstsLocStrings -LiteralPath $PSScriptRoot/module.json
+
+function Add-Tls12InSession {
+    [CmdletBinding()]
+    param()
+
+    try {
+        if ([Net.ServicePointManager]::SecurityProtocol -notcontains 'Tls12') {
+            [Net.ServicePointManager]::SecurityProtocol += [Net.SecurityProtocolType]3072
+            Write-Host (Get-VstsLocString -Key TLS12AddedInSession)
+        }
+        else {
+            Write-Verbose 'TLS 1.2 already present in session.'
+        }
+    }
+    catch {
+        Write-VstsTaskError "Failed to add TLS 1.2 in session: $_.Exception.Message"
+    }
+}
+
+function Assert-TlsError {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)] $exception 
+    )
+
+    if ($exception -eq $null)
+    {
+        return
+    }
+
+    $hasWebException = $false
+    $hasIOException = $false
+    $innerException = $exception
+    while ($innerException -ne $null)
+    {
+        if ($innerException.GetType() -eq [System.Net.WebException])
```

Comment:
I thought that the condition for writing the error was: if( typeof(exception) = web and typeof(exception.innerexception) = io )  then print error.
But with the current logic, we'll print an error even if this order is reversed or if there are intermediate exceptions between the 2 types that we are interested in. Is this correct ?

Reply:
Yes that is deliberate. We changed it to this based on some discussions.